### PR TITLE
New: `block-spacing` rule

### LINKED ISSUE #3303: Can't require space inside block curly
I've searched far and wide, through issues and  `spac*`-, `curly`-, and `block*`- related rules, but I couldn't find anything that worked:
Can I make the missing space inside this curly an error?:
`function nop() {return; }`

Am I missing something?
If not, I'll open a feature request.

### PR DESCRIPTION
**Pull request by mysticatea**
Fixes #3303.

### CODE REVIEW

File: tests/lib/rules/block-spacing.js [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+        {code: "(function() {bar();});", options: ["never"]},
+        {code: "(() => {bar();});", ecmaFeatures: {arrowFunctions: true}, options: ["never"]}
+    ],
+    invalid: [
+        // default/always
+        {
+            code: "{foo();}",
+            options: ["always"],
+            errors: [
+                {type: "BlockStatement", line: 1, colmun: 1, message: "Requires a space after \"{\"."},
```
Comment (nzakas) [inline tests/lib/rules/block-spacing.js]: "column" is spelled wrong in all of these tests.

Comment (mysticatea) [inline tests/lib/rules/block-spacing.js]: Oops, thank you!

File: docs/rules/block-spacing.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# Disallow or enforce spaces inside of single line blocks. (block-spacing)
+
+This rule is for spacing style about single line blocks.
```
Comment (ilyavolodin) [inline docs/rules/block-spacing.md]: about -> around. Or probably even "within", since this rule is controlling spacing inside single line blocks.

**Comment by ilyavolodin** [comment]
LGTM other then one documentation issue.

**Comment by mysticatea** [comment]
Thank you, I modified the doc with "within".

**Comment by garborg** [comment]
@mysticatea, thank you, this is great for me. (Also, the testing in the project impressed me.)

**Pull request merged**